script/deploy-atlas.s.sol

```diff
@@ -22,15 +23,28 @@ contract DeployAtlasScript is DeployBaseScript {
         uint256 deployerPrivateKey = vm.envUint("GOV_PRIVATE_KEY");
         address deployer = vm.addr(deployerPrivateKey);
 
-        // Computes the addresses at which AtlasVerification will be deployed
+
+        
+
+        // Computes the addresses at which contracts will be deployed
         address expectedAtlasAddr = vm.computeCreateAddress(deployer, vm.getNonce(deployer) + 1);
         address expectedAtlasVerificationAddr = vm.computeCreateAddress(deployer, vm.getNonce(deployer) + 2);
         address expectedSimulatorAddr = vm.computeCreateAddress(deployer, vm.getNonce(deployer) + 3);
 
+         // Deploy gas calculator for Arbitrum chains after other deployments
+        address l2GasCalculatorAddr = address(0);
+        uint256 chainId = block.chainid;
+
+        if (chainId == 42_161 || chainId == 42_170 || chainId == 421_614) {
+            // Deploy gas calculator for Arbitrum chains
+            l2GasCalculatorAddr = vm.computeCreateAddress(deployer, vm.getNonce(deployer) + 4);
```

Comment:
It might be better to remove the separate deploy script for the gas calculator, as it complicates this main script a bit - e.g. the computed address here would be the address of the deploying contract, not the gas calculator contract.

We may in the future want to add logic to this main deploy script to conditionally deploy a piece, or use the existing deployment on a chain, when deploying a new Atlas contract. E.g. the calculator seems like one that wouldn't change much and could be reused across many Atlas deployments. But that's probably a feature we should add in a future PR, as it applies to other pieces too.

Reply:
The conditional logic to pass correct args for Arbitrum One/Nova can probably be moved to the constructor of the GasCalculator - that way we don't need all that logic in the deploy script, makes it easier to deploy in 1 line by making a new contract in the main deploy script.

Reply:
Done in #445